api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/calculation/pvls/RoutineCalculationTest.java

```diff
@@ -0,0 +1,128 @@
+package org.openmrs.module.eptsreports.reporting.intergrated.calculation.pvls;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.openmrs.Location;
+import org.openmrs.Obs;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.CalculationContext;
+import org.openmrs.calculation.patient.PatientCalculation;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.pvls.RoutineCalculation;
+import org.openmrs.module.eptsreports.reporting.intergrated.calculation.BasePatientCalculationTest;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportConstants.PatientsOnRoutineEnum;
+
+public class RoutineCalculationTest extends BasePatientCalculationTest {
+
+  @Override
+  public PatientCalculation getCalculation() {
+    return Context.getRegisteredComponents(RoutineCalculation.class).get(0);
+  }
+
+  @Override
+  public Collection<Integer> getCohort() {
+    return Arrays.asList(new Integer[] {2, 6, 7, 8, 999, 432});
+  }
+
+  @Override
+  public CalculationResultMap getResult() {
```

Comment:
Please add test cases that cover permutations that cover if the patient is breastfeeding or not.

Also in order to help quickly analysing these tests and to know what scenarios these are covering, please include in the comments for each of these patients, why the routine calculation would return true/false.

Please do the same for the other test classes. We need this because the scenario is defined in the dataset being loaded into the context we should document what test scenarios have been covered.

Reply:
Just to explain a bit more what I mean here - knowing that we want to know what scenario this case covers instead of just this patient's initiation date.

Reply:
i think it's now clearer why i could have chosen breastfeeding to be another test of its own since this scenario is triggerable with setup. the basic scenario test evaluates ADULTCHILDREN. you can only test other criteria through setting it up as params and the our base evaluator handles one at a time